packages/blockchain-ui/src/components/elements/DeploySteps/DeployStepTwo/DeployStepTwo.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { Component } from 'react';
+import { TextInput, FileUploader, Accordion, AccordionItem, Link } from 'carbon-components-react';
+import IPackageRegistryEntry from '../../../../interfaces/IPackageRegistryEntry';
+
+interface IProps {
+    selectedPackage: IPackageRegistryEntry;
+    onDefinitionNameChange: any;
+    onDefinitionVersionChange: any;
+    currentDefinitionName: string;
+    currentDefinitionVersion: string;
+}
+
+class DeployStepTwo extends Component<IProps> {
+
+    constructor(props: Readonly<IProps>) {
+        super(props);
+
+        this.handleDefinitionNameChange = this.handleDefinitionNameChange.bind(this);
+        this.handleDefinitionVersionChange = this.handleDefinitionVersionChange.bind(this);
+
+    }
+
+    handleDefinitionNameChange(data: any): void {
+        this.props.onDefinitionNameChange(data.target.value);
+    }
+    handleDefinitionVersionChange(data: any): void {
+        this.props.onDefinitionVersionChange(data.target.value);
+    }
+
+    render(): JSX.Element {
+        return (
+            <>
+                <div className='bx--row margin-bottom-06'>
+                    <div className='bx--col-lg-10'>
+                        A smart contract definition describes how the selected package will be deployed in this channel.
+                        The package's name and version (where available) are copied across to the definition as defaults.
+                        You may optionally change them.
+                    </div>
+                </div>
+                <div className='bx--row'>
+                    <h5>Smart contract definition</h5>
+                </div>
+                <div className='bx--row margin-bottom-06'>
+                    <div className='bx--col-lg-5 margin-right-06'>
+                        <TextInput id='nameInput' labelText='Definition name' defaultValue={this.props.currentDefinitionName ? this.props.currentDefinitionName : this.props.selectedPackage.name} onChange={this.handleDefinitionNameChange}></TextInput>
+                    </div>
+                    <div className='bx--col-lg-5'>
+                        <TextInput id='versionInput' labelText='Definition version' defaultValue={this.props.currentDefinitionVersion ? this.props.currentDefinitionVersion : this.props.selectedPackage.version} onChange={this.handleDefinitionVersionChange}></TextInput>
```

Comment:
next button isn't disabled if you don't have a value for name or version

Reply:
Added to https://github.com/IBM-Blockchain/blockchain-vscode-extension/issues/1922